simpleviewinc/keg-hub
# Update Repo Dependencies

**[Semver-Status] major** 

> **Important**
> This PR is dependent on [this PR][https://github.com/simpleviewinc/keg-hub/pull/70]
> Please go review that PR first before this one


## Context

* A number of our dependencies are outdated
* We should stay up-to-date as best we can to ensure long term stability
* This will also allow us to resolve any issues with updates, prior to us being required to update

## Goal

* Stay up to date with most recent versions of our dependencies
* Ensure everything still works as expected

## Updates
* `repos/*/package.json`
  * Updated dependency versions
* `repos/keg-core/app.json`
  * Updated config to match `keg-core` repo name

## Testing

**Notes**
* Each docker package below is running with all changes included in this PR
* For all cases where a repo is a dependency of another repo,
  * The repo has been sync'ed into the container, with the dependency updates from this PR
  * The repo was rebuilt to ensure it's using the updates from the PR
  * Example
   * `keg-core` has `keg-components` and `re-theme` as dependecies
   * When the `keg-core` docker package was made
      * Both `keg-components` and `retheme` were synced
       * And both repos had their dependencies installed to include all changes form this repo

**Tests**
* Run each one of these and ensure the package works as expected
  * **Tap** => `keg evf pack run docker.pkg.github.com/simpleviewinc/keg-packages/tap:update-dependencies`
  * **Core** => `keg core pack run docker.pkg.github.com/simpleviewinc/keg-packages/keg-core:expo-39-upgrade`
  * **Components** => `keg dpg run docker.pkg.github.com/simpleviewinc/keg-packages/components:expo-39-upgrade`
  * **ReTheme** => `keg dpg run docker.pkg.github.com/simpleviewinc/keg-packages/retheme:expo-39-upgrade`
* Next pull this PR, and run the tests for each repo
  * You can also just run them inside of each container above
    * But we want to make sure it works for Daniel, when he runs it without docker
  * Pull PR => `keg && keg pr 65`
  * Run all tests at once => `keg hub run test`
  * Run tests based on context => `keg hub run test <context>`
    * Example => `keg hub run test core` _( runs tests just for keg-core )_
  * Ensure all tests pass
